### ISSUE TITLE
Implement list view for FTA trees and Systems

### ISSUE THREAD

**Issue by blcham** [kbss-cvut/fta-fmea-ui#247]
![image](https://github.com/kbss-cvut/fta-fmea-ui/assets/3962237/3acc6d14-0561-4725-9ff8-3cdadd256e72)

A/C:
- [x] Systems screen have list view
- [x] FTA trees screen have list view and it is default option
- [x] it is possible to switch between list and titles view
- [x] it is described which columns of the table were not implemented due to missing data

**Comment by blcham** [comment]
> @blcham And another question, will the table for systems have exactly the same structure?

No, you can assume now that there is "name" of the System, but that's all for now. The most important part to show now are FTA trees.

**Comment by blcham** [comment]
@kostobog please implement easier part of the table and create ticket to do the reset afterwards.

**Comment by blcham** [comment]
@Kasmadei I realized, that i forgot to update the image (i mentioned it on the meeting but did not update the screenshot):
![image](https://github.com/kbss-cvut/fta-fmea-ui/assets/3962237/c128b621-3ae0-4252-9ea8-5b63f6e40c05)

**Comment by Kasmadei** [comment]
@blcham  Is it correct, that there should be a dropdown with filter for all aircraft types ? So after i select some aircraft type, i'll see in table only aircrafts with selected type.

![Image](https://github.com/kbss-cvut/fta-fmea-ui/assets/49252228/5e2e6070-1521-4cdc-9605-a599b52776d8)

**Comment by blcham** [comment]
Yes but it is not part of this ticket -- read in the screenshot "No need to do filtering yet"

**Comment by Kasmadei** [comment]
@blcham I was thinking that you wanted to add some other changes in the table

**Comment by blcham** [comment]
Please do "docker-compose pull". There are new data to be shown in the list from this PR:
https://github.com/kbss-cvut/fta-fmea/pull/84

![image](https://github.com/kbss-cvut/fta-fmea-ui/assets/3962237/13c05de2-2d57-47fd-b69c-896529a7acf3)

Here is the list of newly added stuff:
- system type
- sub-system type
- creation time
- creator
- last modified
- last editor

**Comment by kostobog** [comment]
@Kasmadei @blcham 
PR kbss-cvut/fta-fmea#85 adds failure rate related properties:
- calculated failure rate, 
- FHA based failure rate and
- required failure rate

**Comment by blcham** [comment]
> @Kasmadei @blcham PR [kbss-cvut/fta-fmea#85](https://github.com/kbss-cvut/fta-fmea/pull/85) adds failure rate related properties:
@Kasmadei  make additional PR to have those fields there

**Comment by blcham** [comment]
Moreover, when I tried to run locally i got table from FTA page showing at System page.

**Comment by Kasmadei** [comment]
@kostobog Hi, after BE update looks like there is empty response from endpoints:

GET http://localhost:1235/services/fta-fmea-server/systems/summaries
GET http://localhost:1235/services/fta-fmea-server/faultTrees/summaries

Can you please check it ?

**Comment by Kasmadei** [comment]
@kostobog Hi. I've received the following response from the endpoint:
GET http://localhost:1235/services/fta-fmea-server/faultTrees/summaries
It seems that there aren't enough new properties. Could you please check? ?
{
    "@context": {
        "types": "@type",
        "components": "http://onto.fel.cvut.cz/ontologies/fta-fmea-application/has-part-component",
        "system": "http://onto.fel.cvut.cz/ontologies/fta-fmea-application/is-artifact-of",
        "functions": "http://onto.fel.cvut.cz/ontologies/fta-fmea-application/has-function",
        "name": "http://onto.fel.cvut.cz/ontologies/fta-fmea-application/name",
        "subsystem": "http://onto.fel.cvut.cz/ontologies/fta-fmea-application/is-performed-by",
        "requiredFailureRate": "http://onto.fel.cvut.cz/ontologies/fta-fmea-application/required-failure-rate",
        "uri": "@id",
        "failureModes": "http://onto.fel.cvut.cz/ontologies/fta-fmea-application/has-failure-mode"
    },
    "@graph": [
        {
            "uri": "http://onto.fel.cvut.cz/ontologies/fta-fmea-application/fault-tree/instance1443295929",
            "types": [
                "http://onto.fel.cvut.cz/ontologies/fta-fmea-application/named-entity",
                "http://onto.fel.cvut.cz/ontologies/fta-fmea-application/managed-entity",
                "http://onto.fel.cvut.cz/ontologies/fta-fmea-application/fault-tree"
            ],
            "requiredFailureRate": 4.0E-4,
            "name": "FC 1.1. Tlak v kabině fault event",
            "system": {
                "uri": "http://onto.fel.cvut.cz/ontologies/fta-fmea-application/system/instance510539860",
                "types": [
                    "http://onto.fel.cvut.cz/ontologies/fta-fmea-application/system",
                    "http://onto.fel.cvut.cz/ontologies/fta-fmea-application/item",
                    "http://onto.fel.cvut.cz/ontologies/fta-fmea-application/named-entity",
                    "http://onto.fel.cvut.cz/ontologies/fta-fmea-application/managed-entity"
                ],
                "name": "acm1",
                "failureModes": [],
                "functions": [],
                "components": []
            },
            "subsystem": {
                "uri": "http://onto.fel.cvut.cz/ontologies/fta-fmea-application/sns-component/instance-1061356481",
                "types": [
                    "http://onto.fel.cvut.cz/ontologies/fta-fmea-application/item",
                    "http://onto.fel.cvut.cz/ontologies/fta-fmea-application/named-entity",
                    "http://onto.fel.cvut.cz/ontologies/fta-fmea-application/managed-entity"
                ],
                "name": "Pressurization control",
                "failureModes": [],
                "functions": [],
                "components": []
            }
        },
        {
            "uri": "http://onto.fel.cvut.cz/ontologies/fta-fmea-application/fault-tree/instance22467827479333998578",
            "types": [
                "http://onto.fel.cvut.cz/ontologies/fta-fmea-application/named-entity",
                "http://onto.fel.cvut.cz/ontologies/fta-fmea-application/managed-entity",
                "http://onto.fel.cvut.cz/ontologies/fta-fmea-application/fault-tree"
            ],
            "requiredFailureRate": 0.003,
            "name": "FC_1.3. Tlak v kabině fault event",
            "system": {
                "uri": "http://onto.fel.cvut.cz/ontologies/fta-fmea-application/system/instance510539860",
                "types": [
                    "http://onto.fel.cvut.cz/ontologies/fta-fmea-application/system",
                    "http://onto.fel.cvut.cz/ontologies/fta-fmea-application/item",
                    "http://onto.fel.cvut.cz/ontologies/fta-fmea-application/named-entity",
                    "http://onto.fel.cvut.cz/ontologies/fta-fmea-application/managed-entity"
                ],
                "name": "acm1",
                "failureModes": [],
                "functions": [],
                "components": []
            },
            "subsystem": {
                "uri": "http://onto.fel.cvut.cz/ontologies/fta-fmea-application/sns-component/instance-1061356481",
                "types": [
                    "http://onto.fel.cvut.cz/ontologies/fta-fmea-application/item",
                    "http://onto.fel.cvut.cz/ontologies/fta-fmea-application/named-entity",
                    "http://onto.fel.cvut.cz/ontologies/fta-fmea-application/managed-entity"
                ],
                "name": "Pressurization control",
                "failureModes": [],
                "functions": [],
                "components": []
            }
        }
    ]
}

**Comment by kostobog** [comment]
@Kasmadei 
Some of the properties are missing in the example. Specifically:
- created, creator, last edited, last editor and calculated failure rate. 

I am no sure why there is no FHA based failure rate. I will check today and let you know.

**Comment by Kasmadei** [comment]
@kostobog Hi, any updates ?

**Comment by blcham** [comment]
@Kasmadei please check data from backend most likely it is updated according to last @kostobog comment.

**Comment by blcham** [comment]
@Kasmadei please past here a picture which fields work.

**Comment by Kasmadei** [comment]
@blcham @kostobog looks like i don't have any trees from response: 
GET 
http://localhost:1235/services/fta-fmea-server/faultTrees/summaries
Response: {"@context":{},"@graph":[]}

I did a docker update with previously provided instruction:

* make sure that fork akaene/fta-fmea-ui is synced with kbss-cvut/ftea-fmea-ui
* cd fta-fmea-ui (assuming it is from akaene fork)
* git pull
* cd deploy/internal-auth
* docker-compose pull
* docker-compose down
* docker-compose up

**Comment by blcham** [comment]
@Kasmadei 

![image](https://github.com/kbss-cvut/fta-fmea-ui/assets/3962237/bb283962-c638-4f44-bf06-88c3045d4209)

**Comment by Kasmadei** [comment]
@blcham  Yes, and /systems works correct, i see the same as you, but for /faultTrees endpoint response is empty :/ In an hour i'll check it again

**Comment by kostobog** [comment]
@blcham I tested running `fta-fmea-ui/depoly/internal-auth` and checked services/db-server. It seams that the import script does not work properly. I can see only one trig file is imported from the `shared/db-server/import/fta-fmea/ava`.

@Kasmadei 
I updated the fields returned in the fault tree summary, for more details see #274 (#275 fixing error in the new example data) and kbss-cvut/fta-fmea#90 .

Now you should be able to get all fields necessary for the table except for status:
- created
- creator - should have uri and username
- modified
- lastEditor - should have uri and username  
- system (Aircraft type ) - there should be uri and name of the aircraft 
- subsystem (ATA) - there should be uri and name of the subsystem, the name is composed of the ATA code and the ATA name
- calculatedFailureRate 
- requiredFailureRate 
- fhaBasedFailureRate 

Here is a snippet of one of the summary of one of the fault trees:
```
calculatedFailureRate : 0.10005625498722481
created : "2024-04-01T10:45:00Z"
creator : {uri: "http://onto.fel.cvut.cz/ontologies/fta-fmea-application/user/test-user-1",…}
	types : ["http://xmlns.com/foaf/0.1/Person"]
	uri:"http://onto.fel.cvut.cz/ontologies/fta-fmea-application/user/test-user-1"
	username:"test-user-1"
fhaBasedFailureRate : 0.01
lastEditor : {uri: "http://onto.fel.cvut.cz/ontologies/fta-fmea-application/user/test-user-2",…}
	types : ["http://xmlns.com/foaf/0.1/Person"]
	uri : "http://onto.fel.cvut.cz/ontologies/fta-fmea-application/user/test-user-2"
	username : "test-user-2"
name : "FC 1.1. Tlak v kabině fault event"
requiredFailureRate : 0.0004
subsystem : {uri: "http://onto.fel.cvut.cz/ontologies/fta-fmea-application/sns-component/instance-1061356481",…}
	components : []
	failureModes : []
	functions : []
	name : "21-30-00 - Pressurization control"
	types : ["http://onto.fel.cvut.cz/ontologies/fta-fmea-application/item",…]
	uri : "http://onto.fel.cvut.cz/ontologies/fta-fmea-application/sns-component/instance-1061356481"
system : {uri: "http://onto.fel.cvut.cz/ontologies/fta-fmea-application/system/instance510539860",…}
	components : []
	failureModes : []
	functions : []
	name : "acm1"
	types : ["http://onto.fel.cvut.cz/ontologies/fta-fmea-application/system",…]
	uri : "http://onto.fel.cvut.cz/ontologies/fta-fmea-application/system/instance510539860"
types : ["http://onto.fel.cvut.cz/ontologies/fta-fmea-application/named-entity",…]
uri : "http://onto.fel.cvut.cz/ontologies/fta-fmea-application/fault-tree/instance1443295929"
```

**Comment by kostobog** [comment]
@Kasmadei Until this issue kbss-cvut/fta-fmea-ui#276 is resolved you can deploy the example data to db-server manually. To do so follow first  run internal-auth as usual. Visit localhost:1235/services/db-server and follow the steps: 
1) select fta-fmea repository from the menu
2) import data - in **import** page select the **server files** tab and import the files listed below. To import a file click the **import** button. This will open the import dialog. In the dialog click `import` button. 
- `for fta-fmea/ava/ava-fta-example-associated-with-fha-fault-event--anonymized.trig`,
-  `fta-fmea/ava/sns-partonomy-example--anonymized.trig` and  
- `fta-fmea/ava/user-example.trig`. 
3) in the **Explore** menu click **Graphs overview**. If import was successful you should see the following graphs:
- The default graph	 
- http://onto.fel.cvut.cz/ontologies/fta-fmea-application/fault-tree/instance1443295929	 
- http://onto.fel.cvut.cz/ontologies/fta-fmea-application/fault-tree/instance22467827479333998578	 
- http://context-with-ata-sns-partonomy-of--acm1	 
- http://context-with-general-ATA-SNS-partonomy	 
- http://context-with-ata-sns-partonomy-of--acm2	 
- http://context-with-independent-items	 
- http://context-with-users

![capture_240509_150301](https://github.com/kbss-cvut/fta-fmea-ui/assets/18463762/28f2a7d1-5c91-42e2-8503-f311010f18d8)

**Comment by Kasmadei** [comment]
@kostobog  Hi, tried your solution and for some reason i don't see fta-fmea/ava/user-example.trig file :( 
Docker containers and akaene/fta-fmea-ui was synchronized with kbss. 
Can i do/check something ? 

![Image](https://github.com/kbss-cvut/fta-fmea-ui/assets/49252228/8d981ce7-1776-4272-8cac-ced575b91566)

**Comment by blcham** [comment]
@Kasmadei 

I think you forgot to do `docker-compose build db-server`, but anyway you do not need to import data manually because the issue  https://github.com/kbss-cvut/fta-fmea-ui/issues/276 was already solved.

**Comment by Kasmadei** [comment]
@blcham No, i did not forget about docker-compose build db-server, and it did not help. I'll try today again :)

**Comment by Kasmadei** [comment]
@blcham @kostobog BE updated without any issue, but response from endpoint is still empty:

Response: {"@context":{},"@graph":[]}

curl 'http://localhost:1235/services/fta-fmea-server/faultTrees/summaries' \
  -H 'Accept: application/json, text/plain, */*' \
  -H 'Accept-Language: en-GB,en-US;q=0.9,en;q=0.8' \
  -H 'Authorization: Bearer eyJhbGciOiJIUzUxMiJ9.eyJzdWIiOiJkZXYiLCJyb2xlcyI6WyJST0xFX1VTRVIiXSwiaWF0IjoxNzE1MzU4NDYxLCJleHAiOjE3MTUzODcyNjF9.AzSTs4Z-66xYNw3JLTOAGz827gqAAiOZXSeEEwgxUvS4RhfS_BJ6Zlyj0uzHbjPlaJ4yLjxb4Udj7jb5IUJGDw' \
  -H 'Connection: keep-alive' \
  -H 'Origin: http://localhost:5173' \
  -H 'Referer: http://localhost:5173/' \
  -H 'Sec-Fetch-Dest: empty' \
  -H 'Sec-Fetch-Mode: cors' \
  -H 'Sec-Fetch-Site: same-site' \
  -H 'User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/124.0.0.0 Safari/537.36' \
  -H 'sec-ch-ua: "Chromium";v="124", "Google Chrome";v="124", "Not-A.Brand";v="99"' \
  -H 'sec-ch-ua-mobile: ?0' \
  -H 'sec-ch-ua-platform: "macOS"'

**Comment by Kasmadei** [comment]
@blcham @kostobog FYI, it works good, i can keep going :)

**Comment by Kasmadei** [comment]
@blcham  @kostobog  New props added: https://github.com/kbss-cvut/fta-fmea-ui/pull/279
After i see more data and it length, feels like i'll need to do some minor visual changes in the table. In my screen (13') it can look a bit better.
I think only status props left.

**Comment by blcham** [comment]
The status will be implemented in the following issue: #280 

The implementation needs to be checked, and if it is okay, we can close it.

**Comment by blcham** [comment]
I see the following issues:

![image](https://github.com/kbss-cvut/fta-fmea-ui/assets/3962237/e4d6bbdb-528a-4a51-a613-1647a6869920)

![image](https://github.com/kbss-cvut/fta-fmea-ui/assets/3962237/0a67eefe-e755-46e0-b0df-c35e8a68f63f)
